File: aula4/java/Item.java
public class Item {

    private String nome;
    private int bonus;

    public Item(String nome, int bonus) {
        if (nome == null || nome.trim().isEmpty()) {
            throw new IllegalArgumentException("O nome do item não pode ser vazio.");
        }
        if (bonus < 0) {
            throw new IllegalArgumentException("O bônus não pode ser negativo.");
        }
        this.nome = nome;
        this.bonus = bonus;
    }

    public String getNome() {
        return this.nome;
    }

    public int getBonus() {
        return this.bonus;
    }

    // Ex: "Espada de Aço (+3)"
    public String descricao() {
        return this.nome + " (+" + this.bonus + ")";
    }
}

File: aula4/python/item.py
class Item:
    def __init__(self, nome: str, bonus: int):
        if nome is None or not nome.strip():
            raise ValueError("O nome do item não pode ser vazio.")
        if bonus < 0:
            raise ValueError("O bônus não pode ser negativo.")
        self._nome = nome
        self._bonus = bonus

    @property
    def nome(self) -> str:
        return self._nome

    @property
    def bonus(self) -> int:
        return self._bonus

    def __str__(self) -> str:
        return f"{self._nome} (+{self._bonus})"